notebooks/nessie-iceberg-hive-demo-nba.ipynb

```diff
@@ -0,0 +1,765 @@
+    "============================\n",
+    "\n",
+    "In this Demo we are a data engineer working at a fictional sports analytics blog. In order for the authors to write articles they have to have access to the relevant data. They need to be able to retrieve data quickly and be able to create charts with it.\n",
+    "\n",
+    "We have been asked to collect and expose some information about basketball players. We have located some data sources and are now ready to start ingesting data into our data lakehouse. We will perform the ingestion steps on a Nessie branch to test and validate the data before exposing to the analysts."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Nessie branches (via Nessie CLI)\n",
+    "----------------------------\n",
+    "Once all dependencies are configured, we can get started with ingesting our basketball data into `Nessie` with the following steps:\n",
+    "\n",
+    "- Create a new branch named `dev`\n",
+    "- List all branches\n",
+    "\n",
+    "It is worth mentioning that we don't have to explicitly create a `main` branch, since it's the default branch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "current_ref = create_ref_catalog(\"dev\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "We have created the branch `dev` and we can see the branch with the Nessie `hash` its currently pointing to.\n",
+    "\n",
+    "Below we list all branches. Note that the auto created `main` branch already exists and both branches point at the same `hash`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "!nessie --verbose branch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create tables under dev branch\n",
+    "-------------------------------------\n",
+    "Once we created the `dev` branch and verified that it exists, we can create some tables and add some data.\n",
+    "\n",
+    "We create two tables under the `dev` branch:\n",
+    "- `salaries`\n",
+    "- `totals_stats`\n",
+    "\n",
+    "These tables list the salaries per player per year and their stats per year.\n",
+    "\n",
+    "To create the data we:\n",
+    "\n",
+    "1. switch our branch context to dev\n",
+    "2. create the table\n",
+    "3. insert the data from an existing csv file. This csv file is already stored locally on the demo machine. A production use case would likely take feeds from official data sources\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Creating our demo schema\n",
+    "current_ref.execute('CREATE SCHEMA IF NOT EXISTS nba');\n",
```

Comment:
Can you add a few `print` statements to indicate the progress and when the cell's finished executing. It runs for quite a while.

Reply:
Sure, added few `print` in the cells with queries that are taking sometime.